shared/src/sys/out.rs

```diff
@@ -62,3 +62,22 @@ pub mod crypto {
         pub fault: u32,
     }
 }
+
+pub mod message {
+    use crate::clock::ChainEpoch;
+    use crate::sys::TokenAmount;
+    use crate::{ActorID, MethodNum};
+
+    #[derive(Debug, Copy, Clone)]
+    #[repr(packed, C)]
+    pub struct MessageDetails {
+        pub caller: ActorID,
+        pub receiver: ActorID,
+        pub method_number: MethodNum,
+        pub value_received: TokenAmount,
+        pub curr_epoch: ChainEpoch,
+        pub version: u32,
```

Comment:
This is the network version? I'd change it to `network_version` to be clear.

Reply:
Also, let's put this _last_. That way we can avoid unaligned loads (due to `repr(packed)`).

Really, I'd order this as:

```rust
pub struct MessageDetails {
    pub base_fee: TokenAmount,
    pub circulating_supply: TokenAmount,
    pub value_received: TokenAmount,
    pub caller: ActorID,
    pub receiver: ActorID,
    pub method_number: MethodNum,
    pub network_curr_epoch: ChainEpoch,
    pub version: u32,
}
```

That way everything is nice and aligned.

Reply:
fixed